ir/shared/src/main/scala/org/scalajs/ir/Types.scala

```diff
@@ -278,7 +278,11 @@ object Types {
     case DoubleType  => DoubleLiteral(0.0)
     case StringType  => StringLiteral("")
     case UndefType   => Undefined()
-    case _           => Null()
+
+    case NullType | AnyType | _:ClassType | _:ArrayType | _:RecordType => Null()
```

Comment:
Since we're doing this, we should also disallow `RecordType`. `null` is not a valid value for a `RecordType`.

(or, we should create a `RecordValue` with fields whose values are recursively generated with `zeroOf`)

Reply:
Ah, ah :) So, it seems we have a typing issue here: A recursively generated record value is definitely the right choice here. However, `zeroOf` is typed to return a `Literal`. A `RecordValue` is not typed as `Literal` (and in general *must not* be typed as literal). However, a `RecordValue` with only `Literals` arguably could be considered a literal.

Reply:
Ah! I had forgotten that it's declared as returning a `Literal`.

In the absence of a use case for calling `zerOf(recordType)`, I guess we should then throw. There's at least one place in `OptimizerCore` that expects a `Literal` from `zeroOf`.

Reply:
Ahahahaha, ironically it essentially creates a (pretransformed) zero of a record type:

https://github.com/scala-js/scala-js/blob/4263c9872729f1c13d391a7995aad0453e1bda68/linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala#L2263-L2268